src/mlpack/methods/ann/layer/recurrent_impl.hpp

```diff
@@ -126,6 +127,44 @@ Recurrent<InputDataType, OutputDataType, CustomLayers...>::Recurrent(
   this->network.push_back(recurrentModule);
 }
 
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+size_t Recurrent<InputDataType, OutputDataType, CustomLayers...>::InputShape() const
+{
+  size_t inputShapeStartModule = boost::apply_visitor(InShapeVisitor(), startModule);
+  size_t inputShapeInputModule = boost::apply_visitor(InShapeVisitor(), inputModule);
+  size_t inputShapeFeedbackModule = boost::apply_visitor(InShapeVisitor(), feedbackModule);
+  size_t inputShapeTransferModule = boost::apply_visitor(InShapeVisitor(), transferModule);
```

Comment:
I guess the compiler is going to optimize this out anyway, but do you think we could put each of the lines into the corresponding if block? And maybe use `const size_t` to make it obvious that the parameter will not change.

Reply:
I have made the changes